### ISSUE TITLE
Report more valid errors at once

### ISSUE THREAD

**Issue by suhr** [astampoulis/makam#86]
It seems that Makam follows the “fail on the first error” approach. It would be much nicer if Makam could report many valid errors at once.

**Comment by astampoulis** [comment]
Makes sense. An incremental improvement that would not require a ton of changes to how the core type checking works is to report errors for all definitions, past the first one that fails.

In terms of type-checking a single definition, reporting multiple errors would require significant changes, and is made more difficult by the presence of type-based constant overloading -- it could result in a large number of error messages as a result of a single error.
